### ISSUE TITLE
[FIRRTL] Add folders for IntegerMulOp

### ISSUE THREAD

**Issue by mikeurbach** [llvm/circt#6724]
This is a primitive Property operation, defined by FIRRTL spec section 25.1.2. General folds for signed arbitrary precision integer arithmetic can be applied, but none were defined in the initial PR. This could include things like:

* [ ] integer_mul(a, 1) -> a
* [ ] integer_mul(constantA, constantB) -> constantA * constantB
* [ ] etc...

**Comment by elhewaty** [comment]
InstCombine pass (LLVM) as some nice folds (can we use them) like:

- X * -1 --> 0 - X
- X * (1 << Z) --> X << Z
- abs(X) * abs(Y) --> abs(X * Y)
- abs(X * abs(Y)) --> abs(X * Y)
- -X * -Y --> X * Y
- sqrt(X) * sqrt(Y) --> sqrt(X * Y)

**Comment by mikeurbach** [comment]
Yep, I think a lot of the classic folds for multiplication will apply. Note that FIRRTL doesn't support negate, shift left, abs, or sqrt yet, but some patterns can be matched and folded.

**Comment by elhewaty** [comment]
@mikeurbach I will add these to folds

**Comment by elhewaty** [comment]
Hello @mikeurbach I will work on this.
please assign it to me.

**Comment by elhewaty** [comment]
Also for the test coverage, should I add tests in   `.fir`, or `.mlir`?

**Comment by mikeurbach** [comment]
Awesome, feel free to take it up! I think we should write the test in .mlir, so we're just testing the transformation, instead of also testing the FIRRTL parser as well as the transformation.
